core/io/http_client.cpp

```diff
@@ -117,8 +117,8 @@ Dictionary HTTPClient::_get_response_headers_as_dictionary() {
 		if (sp == -1) {
 			continue;
 		}
-		String key = s.substr(0, sp).strip_edges();
-		String value = s.substr(sp + 1, s.length()).strip_edges();
+		String key = s.left(sp).strip_edges();
+		String value = s.right(-(sp + 1)).strip_edges();
```

Comment:
This feels very hard to read, I don't think it's an improvement readability wise

Reply:
I tend to agree. I think `s.substr(pos)` is the most readable, so I'm checking now if I can optimize substr(pos) by just calling right(-pos) if a second argument isn't passed.